synapse/config/server.py

```diff
@@ -321,9 +329,30 @@ def read_config(self, config, **kwargs):
                 self.presence_router_config,
             ) = load_module(presence_router_config, ("presence", "presence_router"))
 
-        # Whether to update the user directory or not. This should be set to
-        # false only if we are updating the user directory in a worker
-        self.update_user_directory = config.get("update_user_directory", True)
+        # Which worker is responsible for updating the user directory,
+        # None means the main process handles this.
+        self.worker_to_update_user_directory = config.get(
```

Comment:
Could we add a type annotation to document the expected set of values?
something like `Union[None, False, str, ANY_USER_DIRECTORY_WORKER]`

Reply:
The last one isn't a type, so that wouldn't pass mypy

Reply:
Yup, i'll add a comment though

![image](https://user-images.githubusercontent.com/22740616/145237007-6c87805c-cc75-4202-8ed1-a4a1ad19c910.png)

Reply:
That's very true. Elsewhere in the code we appear to create a single entry enum: https://github.com/matrix-org/synapse/blob/develop/synapse/util/caches/cached_call.py#L72

Though if we're going to make an enum we might as well roll the "main process" and "nobody" values into it.

Reply:
A comment is a good alternative!